test/integration/Cardano/Wallet/Network/HttpBridgeSpec.hs

```diff
@@ -95,56 +83,61 @@ spec = do
-            let msg x = "Expected a 'NodeAvailable' failure but got " <> show x
+        it "gets a 'ErrNetworkUnreachable' if bridge isn't up (1)" $ \network -> do
+            let msg x = "Expected a ErrNetworkUnreachable' failure but got " <> show x
             let action = do
                     res <- runExceptT $ networkTip network
                     res `shouldSatisfy` \case
-                        Left (NodeUnavailable _) -> True
+                        Left (ErrNetworkUnreachable _) -> True
                         _ -> error (msg res)
             action `shouldReturn` ()
 
-        it "listen throws exception when bridge isn't up" $ \network -> do
+        it "gets a 'ErrNetworkUnreachable' if bridge isn't up (2)" $ \network -> do
+            let msg x = "Expected a ErrNetworkUnreachable' failure but got " <> show x
+            let tx = sign txEmpty []
             let action = do
-                    listen network actionDummy
-            action `shouldThrow` anyException
+                    res <- runExceptT $ postTx network tx
+                    res `shouldSatisfy` \case
+                        Left (ErrPostTxNetworkUnreachable (ErrNetworkUnreachable _)) ->
+                            True
+                        _ ->
+                            error (msg res)
+            action `shouldReturn` ()
 
     describe "Submitting signed transactions"
         $ beforeAll startBridge $ afterAll closeBridge $ do
-        it "empty tx fails" $ \(_, network) -> do
+        it "empty tx fails (1)" $ \(_, network) -> do
             let signed = sign txEmpty []
-            (Left err) <- runExceptT $ postTx network signed
-            (show err) `shouldContain`
-                "Transaction failed verification: transaction has no inputs"
+            let err = Left $ ErrPostTxBadRequest
+                    "Transaction failed verification: transaction has no inputs"
+            runExceptT (postTx network signed) `shouldReturn` err
 
-        it "empty tx fails 2" $ \(_, network) -> do
+        it "empty tx fails (2)" $ \(_, network) -> do
             let signed = sign txEmpty [pkWitness]
-            (Left err) <- runExceptT $ postTx network signed
-            (show err) `shouldContain`
-                "Transaction failed verification: transaction has no inputs"
+            let err = Left $ ErrPostTxBadRequest
+                    "Transaction failed verification: transaction has no inputs"
+            runExceptT (postTx network signed) `shouldReturn` err
 
         it "old tx fails" $ \(_, network) -> do
             let signed = sign txNonEmpty [pkWitness]
-            (Left err) <- runExceptT $ postTx network signed
-            (show err) `shouldContain`
-                "Failed to send to peers: Blockchain protocol error"
+            let err = Left $ ErrPostTxProtocolFailure
+                    "Failed to send to peers: Blockchain protocol error"
```

Comment:
As said in slack: if this represents an submitting an alredy-confirmed tx, it would be nice to have a proper case for it in the future.